# HG changeset 9707 patch
From: kaf24@firebug.cl.cam.ac.uk
# Node ID c5fe7d5c4c818ae065d5804c6aa347aa4ec6816f
# Parent  9dda8050d46df88d86f09772b9abf619a6adb475
Subject: Fix cpu_possible_map initialisation.
Signed-off-by: Keir Fraser <keir@xensource.com>
xen-unstable changeset:   10248:7f219d68e6849b873fbd76e9a30b5ba297acbcea
xen-unstable date:        Thu Jun  1 19:14:42 2006 +0100

Acked-by: Jan Beulich <jbeulich@novell.com>

Index: head-2006-06-06/drivers/xen/core/smpboot.c
===================================================================
--- head-2006-06-06.orig/drivers/xen/core/smpboot.c	2006-06-06 12:52:09.000000000 +0200
+++ head-2006-06-06/drivers/xen/core/smpboot.c	2006-06-06 12:52:11.000000000 +0200
@@ -97,9 +97,8 @@ void __init prefill_possible_map(void)
 
 	for (i = 0; i < NR_CPUS; i++) {
 		rc = HYPERVISOR_vcpu_op(VCPUOP_is_up, i, NULL);
-		if (rc == -ENOENT)
-			break;
-		cpu_set(i, cpu_possible_map);
+		if (rc >= 0)
+			cpu_set(i, cpu_possible_map);
 	}
 }
 
